compiler/src/dotty/tools/dotc/transform/MegaPhase.scala

```diff
@@ -221,171 +221,203 @@ class MegaPhase(val miniPhases: Array[MiniPhase]) extends Phase {
 
     def transformNamed(tree: Tree, start: Int, outerCtx: Context): Tree = tree match {
       case tree: Ident =>
-        given Context = prepIdent(tree, start)(using outerCtx)
-        goIdent(tree, start)
+        inContext(prepIdent(tree, start)(using outerCtx)) {
+          goIdent(tree, start)
+        }
```

Comment:
A digression: the pattern seems to suggest a useful addition to stdlib:

```Scala
inline def use[T, U](o: T)(inline op: T ?=> U): U = op(using o)
```

Reply:
Yes, I agree.

Reply:
If we think in that direction, it seems `use` can be used to replace `given` for some use cases:

```Scala
val ctxInner = prepIdent(tree, start)(using outerCtx)
use ctxInner:
  goIdent(tree, start)
```

Reply:
yes, but InContext is a bit clearer for reading, I think.